Refactor the subsampling module to unify the behaviour of equilibrium_detection and statistical_inefficiency (#218)
alchemistry/alchemlyb: docs/preprocessing/alchemlyb.preprocessing.subsampling.rst

.. automodule:: alchemlyb.preprocessing.subsampling

The functions featured in this module can be used to easily subsample either
:ref:`dHdl <dHdl>` or :ref:`u_nk <u_nk>` datasets to give less correlated
timeseries.

High-level functions
--------------------
Two high-level functions
:func:`~alchemlyb.preprocessing.subsampling.decorrelate_u_nk` and

> :func:`~alchemlyb.preprocessing.subsampling.decorrelate_dhdl` could be used to

Reviewer: ```suggestion
:func:`~alchemlyb.preprocessing.subsampling.decorrelate_dhdl` can be used to
```

> preprocess the :ref:`dHdl <dHdl>` or :ref:`u_nk <u_nk>` in a automatic

Reviewer: ```suggestion
preprocess the :ref:`dHdl <dHdl>` or :ref:`u_nk <u_nk>` in an automatic
```

> fashion. The following code could remove the initial burnin period and
> decorrelate the data. ::

Reviewer: ```suggestion
fashion. The following code removes an initial "burnin" period and
decorrelates the data. ::
```
